### PR TITLE
add PoolDataLoader for parrots

### PR DESCRIPTION
**Pull request by jiaoml1996**
In order to increase the speed, add PoolDataLoader for parrots.

### CODE REVIEW

**Comment by jiaoml1996** [comment]
In order to increase the speed, add PoolDataLoader for parrots.

File: mmedit/datasets/builder.py [hunk 1]
```diff
@@ -4,15 +4,22 @@
...
-from torch.utils.data import ConcatDataset, DataLoader
+from torch.utils.data import ConcatDataset
 
 from .dataset_wrappers import RepeatDataset
 from .registry import DATASETS
 from .samplers import DistributedSampler
 
+if torch.__version__ == 'parrots':
+    from torch.utils.data import PoolDataLoader
+    DataLoader = partial(PoolDataLoader, prefetch_num=2)
```
Comment (nbei) [inline mmedit/datasets/builder.py:19]: The `prefetch_num` may be set by config.

File: mmedit/datasets/builder.py [hunk 2]
```diff
@@ -133,6 +140,8 @@ def build_dataloader(dataset,
         worker_init_fn, num_workers=num_workers, rank=rank,
         seed=seed) if seed is not None else None
 
+    if torch.__version__ == 'parrots':
+        pin_memory = False
```
Comment (nbei) [inline mmedit/datasets/builder.py:144]: `pin_memory` can also be set by config.

**Comment by OpenMMLab-Assistant-007** [comment]
Hi!
@magicdream2222 
First of all, we want to express our gratitude for your significant PR in the OpenMMLab project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**